src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1323,6 +1323,57 @@ bool SQLStorage::loadDeviceInstallationResult(data::InstallationResult* result,
   return true;
 }
 
+void SQLStorage::saveEcuReportCounter(const Uptane::EcuSerial& ecu_serial, const int64_t counter) {
+  SQLite3Guard db = dbConnection();
+
+  auto statement = db.prepareStatement<std::string, int64_t>(
+      "INSERT OR REPLACE INTO ecu_report_counter (ecu_serial, counter) VALUES "
+      "(?,?);",
+      ecu_serial.ToString(), counter);
```

Comment:
I'm not sure `INSERT OR REPLACE` is right here. We should enforce one counter per ECU, so won't we need a `WHERE` clause? The counter itself doesn't need to be unique.

Reply:
I didn't use `WHERE` because I also use this function in initializer, when it's an empty table. After initialization, every call of this function should use an existing ecu_serial from database, so there will be only one line for each ecu in this ecu_report_counter table.

Reply:
Okay, I think this was my mistake. I'm out of practice with SQL! Looking at other examples, this looks fine.